config/prometheus/templates/deployment.yaml

```diff
@@ -59,10 +60,17 @@ spec:
           timeoutSeconds: 5
         resources:
           {{- toYaml .Values.kubeStateMetrics.resources | nindent 10 }}
+        {{- if .Values.promServer.kubeconfigSecret }}
+        volumeMounts:
+        - name: kubeconfig-volume
+          mountPath: /.kube
```

Comment:
Unsure if root's home here is `/` or `/root`
may also want to add in `automountServiceAccountToken: false` on the pod spec if we aren't going to use the built in service account